app/scripts/modules/core/src/pipeline/service/execution.service.ts

```diff
@@ -240,13 +240,17 @@ export class ExecutionService {
   public startAndMonitorPipeline(app: Application, pipeline: string, trigger: any): IPromise<IRetryablePromise<void>> {
     const { executionService } = ReactInjector;
     return PipelineConfigService.triggerPipeline(app.name, pipeline, trigger).then(triggerResult =>
-      executionService.waitUntilTriggeredPipelineAppears(app, triggerResult),
+      executionService.waitUntilTriggeredPipelineAppears(app, pipeline, triggerResult),
     );
   }
 
-  public waitUntilTriggeredPipelineAppears(application: Application, eventId: string): IRetryablePromise<any> {
+  public waitUntilTriggeredPipelineAppears(
```

Comment:
Can we also limit the retry period here to something reasonable?

Reply:
Yeah, I'm working on another PR that will either limit the retry period, or move the retry to look in the execution data source already in the front end rather than hitting the back end again.

Reply:
👍

Reply:
#6707